File: backend/app/schemas/summary.py
from app.schemas import base as schema_base
from app.schemas import company as schema_company


class SummaryAmount(schema_base.ORMBaseModel):
    rid: int
    date_snap: str
    rank: int
    quarter1_expected: int
    quarter1_order: int
    quarter2_expected: int
    quarter2_order: int
    quarter3_expected: int
    quarter3_order: int
    quarter4_expected: int
    quarter4_order: int
    half_first_expected: int
    half_first_order: int
    half_second_expected: int
    half_second_order: int
    all_expected: int
    all_order: int


class SummaryCompany(schema_base.ORMBaseModel):
    rid: int
    date_snap: str
    company: schema_company.Company
    rank: int
    quarter1_expected: int
    quarter1_order: int
    quarter2_expected: int
    quarter2_order: int
    quarter3_expected: int
    quarter3_order: int
    quarter4_expected: int
    quarter4_order: int
    half_first_expected: int
    half_first_order: int
    half_second_expected: int
    half_second_order: int
    all_expected: int
    all_order: int


class SummaryCreate(schema_base.ORMBaseModel):
    date_snap: str


class SankeyCompany(schema_base.ORMBaseModel):
    rid: int
    name: str
    amount: int


class SankeyProjectGroup(schema_base.ORMBaseModel):
    rid: int
    name: str
    company_rid: int
    company_name: str
    amount: int
    project_count: int


class SankeyProjectCount(schema_base.ORMBaseModel):
    rid: int
    project_count: int


class SankeyCompanyPm(schema_base.ORMBaseModel):
    company_rid: int
    company_name: str
    project_group_rid: int
    project_group_name: str
    pm_rid: int
    pm_name: str
    amount: int


class SankeyPmPl(schema_base.ORMBaseModel):
    pm_rid: int
    pm_name: str
    pl_rid: int
    pl_name: str
    project_group_rid: int
    project_group_name: str
    amount: int


class SankeySummary(schema_base.ORMBaseModel):
    year: int
    total_amount: int
    companies: list[SankeyCompany]
    project_groups: list[SankeyProjectGroup]
    company_project_counts: list[SankeyProjectCount]
    pm_project_counts: list[SankeyProjectCount]
    pl_project_counts: list[SankeyProjectCount]
    company_pm: list[SankeyCompanyPm]
    pm_pl: list[SankeyPmPl]

File: frontend/app/src/types/Summary.ts
import type { Company } from '@/types/Company'
import { TypeRank } from '@/types/Project'

export interface SummaryAmount {
  rid: number
  date_snap: string
  rank: TypeRank
  quarter1_expected: number
  quarter1_order: number
  quarter2_expected: number
  quarter2_order: number
  quarter3_expected: number
  quarter3_order: number
  quarter4_expected: number
  quarter4_order: number
  half_first_expected: number
  half_first_order: number
  half_second_expected: number
  half_second_order: number
  all_expected: number
  all_order: number
}

export interface SummaryCompany {
  rid: number
  date_snap: string
  company: Company
  rank: number
  quarter1_expected: number
  quarter1_order: number
  quarter2_expected: number
  quarter2_order: number
  quarter3_expected: number
  quarter3_order: number
  quarter4_expected: number
  quarter4_order: number
  half_first_expected: number
  half_first_order: number
  half_second_expected: number
  half_second_order: number
  all_expected: number
  all_order: number
}

export interface SankeyCompany {
  rid: number
  name: string
  amount: number
}

export interface SankeyProjectGroup {
  rid: number
  name: string
  company_rid: number
  company_name: string
  amount: number
  project_count: number
}

export interface SankeyProjectCount {
  rid: number
  project_count: number
}

export interface SankeyCompanyPm {
  company_rid: number
  company_name: string
  project_group_rid: number
  project_group_name: string
  pm_rid: number
  pm_name: string
  amount: number
}

export interface SankeyPmPl {
  pm_rid: number
  pm_name: string
  pl_rid: number
  pl_name: string
  project_rid: number
  project_name: string
  project_group_rid: number
  amount: number
}

export interface SankeySummary {
  year: number
  total_amount: number
  companies: SankeyCompany[]
  project_groups: SankeyProjectGroup[]
  company_project_counts: SankeyProjectCount[]
  pm_project_counts: SankeyProjectCount[]
  pl_project_counts: SankeyProjectCount[]
  company_pm: SankeyCompanyPm[]
  pm_pl: SankeyPmPl[]
}
